platforms/android/src/jp/wizcorp/phonegap/plugin/WizAssets/WizAssetManager.java

```diff
@@ -197,4 +198,34 @@ public void deleteFolder(String filePath) {
             Log.e(TAG, "Delete file error -- " + e.getMessage(), e);
         }
     }
+
+    // If we detect a data structure created by plugin version <= 5.0.0 we want to clean up
+    private void checkForMigration() {
+        String deprecatedDbLocation = that.getCacheDir().getAbsolutePath() + File.separator + "assets.db";
+        File deprecatedDbFile = new File(deprecatedDbLocation);
+        if (!deprecatedDbFile.exists()) {
+            return;
+        }
+
+        Log.d(TAG, "Found a deprecated database: removing cached files and database");
+        SQLiteDatabase deprecatedDb = SQLiteDatabase.openDatabase(deprecatedDbLocation, null, SQLiteDatabase.OPEN_READONLY);
+        int counter = 0;
+        try {
+            Cursor cursor = deprecatedDb.rawQuery("select * from assets", null);
+            String filePath;
+            while (cursor.moveToNext()) {
+                filePath = cursor.getString(cursor.getColumnIndex("filePath"));
+                File fileToDelete = new File(filePath);
+                if (fileToDelete.exists() && fileToDelete.delete()) {
```

Comment:
Same as `createFolderIfRequired`, could it be possible to invert the two? Such as:
`fileToDelete.delete() || !fileToDelete.exists()`